tests/buffer/buffer_constructors_common.h

```diff
@@ -27,6 +27,42 @@
 namespace buffer_constructors_common {
 using namespace sycl_cts;
 
+template <typename Acc>
+struct iota_kernel {
+  Acc acc1, acc2;
+  void operator()(sycl::id<1> id) const {
+    acc1[id] = static_cast<typename Acc::value_type>(id[0]);
+    acc2[id] = static_cast<typename Acc::value_type>(id[0]);
+  }
+};
+
+inline std::stringstream alloc_log;
+
+template <typename T>
+class logging_alloc {
+ public:
+  typedef T value_type;
+  typedef T *pointer;
+  typedef size_t size_type;
```

Comment:
1. Changed.

2. [`value_type`](https://github.com/intel/llvm/blob/3c8cf0b013f4908ad31a931d984c448c58b74d9f/sycl/include/sycl/detail/sycl_mem_obj_allocator.hpp#L41) I've not found reference to `pointer` and `size_type` in Intel LLVM (the only implementation supporting the buffer test suite), but others may. The spec says about `sycl::buffer_allocator<T>`:
> It is the default buffer allocator used by the runtime, when no allocator is defined by the user. Meets the C++ named requirement [Allocator](https://en.cppreference.com/w/cpp/named_req/Allocator).

If you ask me, even this allocator is incomplete and may even be insufficient for an otherwise conforming implementation that wants to use the other absent type aliases. If anything, I didn't go the extra mile and implement the missing ones, just reused `custom_alloc` from `buffer_storage_common.h`.

Reply:
Ah... Thanks for clarification. I was looking for uses inside test sources rather than implementation sources.